File: libandroid-navigation-ui/src/test/java/com/mapbox/services/android/navigation/ui/v5/BaseTest.java
package com.mapbox.services.android.navigation.ui.v5;

import com.mapbox.api.directions.v5.models.DirectionsRoute;
import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;

import java.io.IOException;

public class BaseTest {

  protected static final double DELTA = 1E-10;
  protected static final String ACCESS_TOKEN = "pk.XXX";

  private TestRouteBuilder routeBuilder;
  private TestRouteProgressBuilder routeProgressBuilder;

  public BaseTest() {
    routeBuilder = new TestRouteBuilder();
    routeProgressBuilder = new TestRouteProgressBuilder();
  }

  protected String loadJsonFixture(String filename) throws IOException {
    return routeBuilder.loadJsonFixture(filename);
  }

  protected DirectionsRoute buildTestDirectionsRoute() throws IOException {
    return routeBuilder.buildTestDirectionsRoute(null);
  }

  protected RouteProgress buildRouteProgress(DirectionsRoute route,
                                             double stepDistanceRemaining,
                                             double legDistanceRemaining,
                                             double distanceRemaining,
                                             int stepIndex,
                                             int legIndex) throws Exception {
    return routeProgressBuilder.buildTestRouteProgress(
      route, stepDistanceRemaining, legDistanceRemaining, distanceRemaining, stepIndex, legIndex
    );
  }
}

File: libdirections-offboard/src/test/java/com/mapbox/navigation/route/offboard/base/BaseTest.kt
package com.mapbox.navigation.route.offboard.base

import com.google.gson.GsonBuilder
import com.mapbox.api.directions.v5.DirectionsAdapterFactory
import com.mapbox.api.directions.v5.models.DirectionsResponse
import com.mapbox.api.directions.v5.models.DirectionsRoute
import java.io.IOException
import java.util.Scanner
import okhttp3.internal.Util.UTF_8

open class BaseTest {

    companion object {
        private const val MULTI_LEG_ROUTE_FIXTURE = "directions_two_leg_route.json"
    }

    @Throws(IOException::class)
    protected fun loadJsonFixture(filename: String): String {
        val inputStream = javaClass.classLoader?.getResourceAsStream(filename)
        return inputStream?.let {
            val scanner = Scanner(it, UTF_8.name()).useDelimiter("\\A")
            if (scanner.hasNext()) scanner.next() else ""
        } ?: ""
    }

    @Throws(Exception::class)
    protected fun buildMultipleLegRoute(): DirectionsRoute {
        val body = loadJsonFixture(MULTI_LEG_ROUTE_FIXTURE)
        val gson = GsonBuilder().registerTypeAdapterFactory(DirectionsAdapterFactory.create()).create()
        val response = gson.fromJson(body, DirectionsResponse::class.java)
        return response.routes()[0]
    }
}
